### PR TITLE
Switch to coredns with DNSendpoint plugin

### PR DESCRIPTION
**Pull request by k0da**
* put dnstype as label
* Drop etcd, internal extdns instance, and coredns with etcd plugin
* Use CoreDNS with DNSEndpoint plugin

Signed-off-by: Dinar Valeev <dinar.valeev@absa.africa>

### CODE REVIEW

**Review by somaritane — APPROVED** [review]
lgtm 👍

**Review by kuritka — APPROVED** [review]

**Comment by k0da** [comment]
@ytsarev thoughts?

**Comment by k0da** [comment]
image (6)

**Review by kuritka — DISMISSED** [review]

**Review by ytsarev — COMMENTED** [review]
It looks super cool - but how we test this shotgun-surgery level change? With current terratest test suite almost as it is.
@k0da please clarify the test automation approach

File: controllers/dnsupdate.go [hunk 1]
```diff
@@ -94,6 +94,7 @@ func (r *GslbReconciler) gslbDNSEndpoint(gslb *k8gbv1beta1.Gslb) (*externaldns.D
 			Name:        gslb.Name,
 			Namespace:   gslb.Namespace,
 			Annotations: map[string]string{"k8gb.absa.oss/dnstype": "local"},
+			Labels:      map[string]string{"k8gb.absa.oss/dnstype": "local"},
```
[the remark this answers is not in the document]
Comment (k0da) [reply controllers/dnsupdate.go:97]: Yes for backward compatibilty.

**Comment by k0da** [comment]
@ytsarev terratest brings coredns up, terratest runs dig against ```new``` CoreDNS and receives expected result from it. 

Change isn't huge. We just cutted of ETCD as medium for CoreDNS and use DNSEndpoint object directly.

**Comment by ytsarev** [comment]
@k0da could you point me to the tests you are talking about?

**Comment by k0da** [comment]
@ytsarev https://github.com/AbsaOSS/k8gb/blob/master/terratest/test/k8gb_basic_failover_test.go#L56 for example

**Comment by ytsarev** [comment]
@k0da so pretty much existing regression test suite. Ok, it makes sense

**Review by ytsarev — APPROVED** [review]

**Review by kuritka — APPROVED** [review]

**Review by somaritane — APPROVED** [review]

**Pull request merged**